### PR TITLE
(TWILL-230) Get resource report based on the caller user

### PR DESCRIPTION
**Pull request by chtyim**
- Also by default get the resource report from the tracking url, then fall back to the original tracking url.

### CODE REVIEW

File: twill-yarn/src/main/java/org/apache/twill/yarn/YarnTwillController.java [hunk 1]
```diff
@@ -322,7 +314,46 @@ private boolean hasRun(YarnApplicationState state) {
...
   }
+
+  /**
+   * Returns the {@link ResourceReportClient} for fetching resource report from the AM.
+   * It first consults the RM for the tracking URL and get the resource report from there.
+   */
+  @Nullable
+  private ResourceReportClient getResourcesClient() {
+    YarnApplicationReport report = processController.getReport();
+    List<URL> urls = new ArrayList<>(2);
```
Comment (yaojiefeng) [inline twill-yarn/src/main/java/org/apache/twill/yarn/YarnTwillController.java]: Why will this url list has initial capacity of 2?

Comment (chtyim) [inline twill-yarn/src/main/java/org/apache/twill/yarn/YarnTwillController.java]: Because we store the tracking URL and the original tracking URL.

**Comment by anwar6953** [comment]
LGTM

**Comment by anwar6953** [comment]
LGTM

**Pull request closed without merging**